### ISSUE TITLE
Watch always returns watch_id=0

### ISSUE THREAD

**Issue by yehiyam** [mixer/etcd3#61]
For the cancel watch to work, the watch_id field needs to be saved.
From what I can see, the watch_id is always 0 (the default value?). This causes the cancelWatch gRPC method to always have id 0.
It seems that watchResponse message is too short. It only decodes the messageHeader
I'm not familiar with gRPC so I might be missing something.
Could you take a look?
Thanks

**Comment by yehiyam** [comment]
I figured out that this happens when the sequence is:
- watch
- cancel
- watch
- cancel
- ...
I saw in the code that the client will close the stream after the cancel and create a new one for the new watch. I assume that the server creates a new object with new indexing.

**Comment by reptilbud** [comment]
Hey, I had the same problem !
All my watchers ids were 0 causing trouble when canceling them.
Maybe it's not the same cause than yours, but in case of here what I did:
- Created an instance of the library
- Based on this instance, created a namespace object
- Based on this namespace, created some watchers
I was assuming that I used the same "namespace" instance in my all app but in fact I did a little mistake, and the namespace was not a singleton but it was created each time causing the new watchers to have the id "0".

Maybe check if you're not re-creating a new instance of the lib/namespace to be sure all your watchers are linked to the same instance.

Hope it helps
Cheers
